inference-engine/src/mkldnn_plugin/nodes/mkldnn_scatter_update_node.cpp

```diff
@@ -0,0 +1,498 @@
+    SizeVector indicesDim = getParentEdgeAt(INDICES_ID)->getDesc().getDims();
+    size_t srcRank = srcDataDim.size();
+    int axis = 0;
+    if (axisRelaxed) {
+        auto &axisMemPtr = getParentEdgeAt(AXIS_ID)->getMemoryPtr();
+        uint8_t *axisPtr = reinterpret_cast<uint8_t*>(axisMemPtr->GetData()) +
+            axisMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding * axisSize;
+        if (axisSize == 4) {
+            auto *axisPtr32 = reinterpret_cast<int32_t*>(axisPtr);
+            axis = *axisPtr32;
+        } else {
+            auto *axisPtr64 = reinterpret_cast<int64_t*>(axisPtr);
+            axis = *axisPtr64;
+        }
+
+        if (axis >= static_cast<int>(srcRank) || axis < (static_cast<int>(srcRank) * - 1)) {
+            THROW_IE_EXCEPTION << "'" << getType() << "'" << " layer with name '" << getName()
+            << "' should have axis value in range [-r, r - 1], where r is the rank of input data";
+        }
+        axis = axis < 0 ? (axis + srcRank) : axis;
+
+        size_t srcDimAxis = srcDataDim[axis];
+        std::vector<size_t> indicesBlockND = getBlockND(indicesDim);
+        parallel_nt(0, [&](const int ithr, const int nthr) {
+            size_t start = 0, end = 0;
+            splitter(indicesBlockND[0], nthr, ithr, start, end);
+            for (int i = start; i < end; i++) {
+                int64_t idxValue =  getIndicesValue(indicesPtr, i);
+                if (idxValue >= static_cast<int64_t>(srcDimAxis) || idxValue < 0) {
+                    THROW_IE_EXCEPTION << "'" << getType() << "'" << " layer with name '" << getName()
+                    << "' have indices value that points to non-existing output tensor element";
+                }
+            }
+        });
+
+        if (scatterUpdateMode == ScatterUpdateMode::ScatterUpdate) {
+            SizeVector indicesDim = getParentEdgeAt(INDICES_ID)->getDesc().getDims();
+            SizeVector updateDim = getParentEdgeAt(UPDATE_ID)->getDesc().getDims();
+            size_t indicesRank = indicesDim.size();
+            size_t updateRank = updateDim.size();
+            SizeVector expectUpdateShape = {};
+            for (size_t rs = 0; rs < srcRank; rs++) {
+                if (rs != axis) {
+                    expectUpdateShape.push_back(srcDataDim[rs]);
+                } else {
+                    for (size_t ri = 0; ri < indicesRank; ri++) {
+                        expectUpdateShape.push_back(indicesDim[ri]);
+                    }
+                }
+            }
+            for (size_t ru = 0; ru < updateRank; ru++) {
+                if (updateDim[ru] != expectUpdateShape[ru]) {
+                    THROW_IE_EXCEPTION << "'" << getType() << "'" << " layer with name '" << getName()
+                    << "' do not have matched tensor shape relationship for input, indices and update";
+                }
+            }
+        }
+    }
+
+    if (srcPtr != dstPtr) {
```

Comment:
In fact right this condition will always be true, so I am wondering does it make sense to enable "inplace" feature for this layer? So smt like:
`
config.inConfs[DATA_ID].inPlace = 0;
config.outConfs[0].inPlace = 0;
`
This will allow src and dst pointers to point on the same physical memory and could provide some performance benefit I suppose.

Reply:
Make muck sense. thank you Dmitry. fixed.